vmware-event-router/internal/provider/vcenter/vcenter.go

```diff
@@ -180,12 +184,11 @@ func (vc *EventStream) Stream(ctx context.Context, p processor.Processor) error
 		if ctx.Err() != nil {
 			ctx = context.Background()
 		}
-		err = ec.Destroy(ctx)
-		if err != nil {
-			vc.Errorf("could not destroy property collector: %v", err)
-		}
+		_ = ec.Destroy(ctx) // ignore any err
```

Comment:
The issue is that `Destroy` will likely return with error due to already being logged out from the stream returning above due to context.Cancelled. So this error is actually most likely a false positive. Also the error itself is not critical (optimization and safety net for logout if stream does not do that automatically when the context is cancelled and the underlying connection is terminated).

I had another version we I captured the destroy error in a local variable and printed it because otherwise I would shadow `err` on return and this processor would always incorrectly error out on shutdown, leading to exit code != 0.

Reply:
Ah, ok this sounds reasonable, those defers and multiple contexts can be confusing to track down and review to be honest and you need to keep a stack of what is called when in your brain which can error out (in my mind like how it happened here). This makes me think how to order the code in a way that is more readable even though we have those layers. This is not a problem, just a thing that I am wondering about ATM and I don't have good suggestions now. We can discuss this offline. Thanks again for the input :beers: